libs/spandrel/spandrel/architectures/PLKSR/__arch/RealPLKSR.py

```diff
@@ -134,8 +144,20 @@ def __init__(
             torch.repeat_interleave, repeats=upscaling_factor**2, dim=1
         )
 
-        self.to_img = nn.PixelShuffle(upscaling_factor)
+        if dysample and upscaling_factor != 1:
+            groups = out_ch if 3 * upscaling_factor**2 < 4 else 4
+            self.to_img = DySample(
+                in_ch * upscaling_factor**2,
+                out_ch,
+                upscaling_factor,
+                groups=groups,
+                end_convolution=upscaling_factor != 1,
+            )
```

Comment:
I don't get this code. Since `upscaling_factor >= 1` is always true, this condition ensures that `upscaling_factor >= 2`. But this means that `group` will always be `4`, since `3 * upscaling_factor**2 < 4` can never be true. Similarly, `end_convolution` will always be true.

Did you mess up copying the original code, or was the original this messy?

Reply:
[this is straight from the original](https://github.com/muslll/neosr/blob/a8fe55606c15eb8a5c1abdbb4c7913435f968b20/neosr/archs/realplksr_arch.py#L143)

Reply:
... Then let's just keep it as is.